### DISCUSSION TITLE
What does PHOTOPRISM_DISABLE_HEIFCONVERT do?

### DISCUSSION THREAD

**Discussion by larsxschneider** [photoprism/photoprism#2763] [Q&A] [answered]
I assume it disables the conversion from HEIF files to JPG 😄 

But does that mean HEIF files are ignore completely? Or does it mean thumbnails and low-res images are generated for HEIF files but not original sized sidecar files?

**Comment by lastzero** [comment, marked as answer]
It breaks the conversation so you don't even get small thumbnails unless you convert the HEIF files manually or using another tool, so PhotoPrism can pick up those JPEG files.

**Reply by larsxschneider** [reply]
Thank you! That makes sense! If I set `PHOTOPRISM_JPEG_SIZE` to `720`. Does this have any negative consequences for face detection etc? My goal is just to safe space with the sidecar files. Currently, I need twice the amount of space (HEIF originals + JPEG in sidecar).

I am even contemplating to ditch HEIF and go back to JPEG again 🤷

**Reply by larsxschneider** [reply]
Plus, would you welcome a PR to add the info you have given me to the `photoprism help` command or do you want to keep that command output brief?

**Reply by lastzero** [reply]
Might make more sense in the docs as few people read the command help. Space there also is limited. 720 will work for face detection if you don't care about very small faces, but recognition generally is much better with 1280.

**Reply by larsxschneider** [reply]
Thank you! I'll update the docs.

Is there value in keeping the side car files at full resolution (`<7680` pixels) for face detection? IOW is the full resolution used for face detection?

Face detection is such an amazing feature and I want to make sure the algorithm can perform best.

**Reply by lastzero** [reply]
At least the way we do face detection, it would BE pretty slow with large images like originals. Also it would find faces in tiny background details most humans probably wouldn't notice,  especially with modern cameras that have a lot of MP.

**Reply by larsxschneider** [reply]
But that's the default, no? By default `<7680px` JPG files are generated as side car files and if I understand you correctly, then those are _currently_ used for face detection, no?

**Reply by lastzero** [reply]
Not for detection but for recognition, which is the step after detection. At that time, we know how large a face is, so the 160px crop is extracted from a suitable thumbnail. If the image is 7000px and the crop 160px, then the face would be super tiny. So this won't happen in practice.
